crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -172,79 +578,111 @@ impl<F> TryFrom<&BankofamericaRouterData<&types::RefundsRouterData<F>>>
         item: &BankofamericaRouterData<&types::RefundsRouterData<F>>,
     ) -> Result<Self, Self::Error> {
         Ok(Self {
-            amount: item.amount.to_owned(),
+            order_information: OrderInformation {
+                amount_details: Amount {
+                    total_amount: item.amount.clone(),
+                    currency: item.router_data.request.currency.to_string(),
+                },
+            },
+            client_reference_information: ClientReferenceInformation {
+                code: Some(item.router_data.request.refund_id.clone()),
+            },
         })
     }
 }
 
-// Type definition for Refund Response
-
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
-}
-
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
+impl From<BankofamericaPaymentStatus> for enums::RefundStatus {
+    fn from(item: BankofamericaPaymentStatus) -> Self {
```

Comment:
Why are we using same status type for payments and refunds? Please create new enum type

Reply:
Done